### PR TITLE
[MRG] Fix MPIBackend platform logic

### LINKED ISSUE #875: MPIBackend platform logic
Was inspecting the MPIBackend class and found this logic below.

https://github.com/jonescompneurolab/hnn-core/blob/ccef332d3190a6a7338703c74641abd817d3056d/hnn_core/parallel_backends.py#L650-L655

Isn't this setting posix=True for Windows and Mac (Dar**win**), and False for Linux? [sys.platform docs](https://docs.python.org/3/library/sys.html#sys.platform)

Although somehow MPI seems to be working OK on Linux despite the incorrect flag here?

### PR DESCRIPTION
**Pull request by gtdang**
Updated the logic for MPIBackend and get_mpi_env function to not use the obtuse 'win' string search. 

closes #875

### CODE REVIEW

File: hnn_core/parallel_backends.py [hunk 1]
```diff
@@ -84,7 +84,8 @@ def _gather_trial_data(sim_data, net, n_trials, postproc):
 def _get_mpi_env():
     """Set some MPI environment variables."""
     my_env = os.environ.copy()
-    if 'win' not in sys.platform:
+    # For Linux systems
+    if sys.platform not in ['win32', 'darwin']:
```
Comment (gtdang) [inline hnn_core/parallel_backends.py:88]: I'm not sure if the original intent of this line was to just target Linux systems, but that was what it was doing using the `'win' not in` logic. This has the same functionality but is more explicit of the platforms it avoids. 

The old logic would have also excluded the 'cygwin' platform, a posix-translator layer for windows... but I doubt anyone will be trying to use hnn-core on that.

File: hnn_core/parallel_backends.py [hunk 2]
```diff
@@ -648,11 +649,7 @@ def __init__(self, n_procs=None, mpi_cmd='mpiexec'):
                          'mpi_child.py')
 
         # Split the command into shell arguments for passing to Popen
-        if 'win' in sys.platform:
-            use_posix = True
-        else:
-            use_posix = False
-        self.mpi_cmd = shlex.split(self.mpi_cmd, posix=use_posix)
+        self.mpi_cmd = shlex.split(self.mpi_cmd, posix=sys.platform != 'win32')
```
Comment (jasmainak) [inline hnn_core/parallel_backends.py:652]: what about `cygwin` as you mention in your comment above?

**Comment by jasmainak** [comment]
what's the motivation for this fix? Does it break something for you? If it ain't broken, don't fix it ... we don't have the means to test it properly

File: hnn_core/parallel_backends.py (hunk 1 above)
Comment (jasmainak) [reply hnn_core/parallel_backends.py:88]: You'll have to look at Blake's old commits to understand why this flag was added. Likely a Windows-specific thing not related to Mac.

Comment (rythorpe) [reply hnn_core/parallel_backends.py:88]: I'm guessing this is intended only for unix systems which support open-mpi (vs. ms-mpi), in which case we should only be avoiding windows here.

Comment (rythorpe) [reply hnn_core/parallel_backends.py:88]: Also, setting the `OMPI_MCA_btl_base_warn_component_unused` environment variable to `0` is only used for suppressing warning messages. Regardless of Blake's original intention here, I suggest that we either implement this consistently for all references to an open-mpi build (i.e., across Mac and Linux systems) or not at all.

File: hnn_core/parallel_backends.py (hunk 2 above)
Comment (gtdang) [reply hnn_core/parallel_backends.py:652]: I don't think people are going to be using Cygwin... but it seems to be able to use posix or win32 style-paths. So to write a fix for that platform, we'd need to be able to detect the style of path the user supplied and use correct parsing flag. It doesn't seem worth the effort unless we know of anyone who uses it...

Comment (jasmainak) [reply hnn_core/parallel_backends.py:652]:
```suggestion
        use_posix = True if sys.platform != 'win32' else False
        self.mpi_cmd = shlex.split(self.mpi_cmd, posix=use_posix)
```

just a bit more explicit and clear

**Review by jasmainak — APPROVED** [review]
tiny nitpick, otherwise LGTM. You need to update the PR from "Draft". @rythorpe merge button is yours

File: hnn_core/parallel_backends.py (hunk 1 above)
Comment (rythorpe) [reply hnn_core/parallel_backends.py:88]: To be clear, I'm suggesting we remove `'darwin'` from the list above or remove L87-89 altogether. Aside from that and @jasmainak's comments, this looks good to go by me.

File: hnn_core/parallel_backends.py (hunk 2 above)
Comment (gtdang) [reply hnn_core/parallel_backends.py]: Thanks!

**Review by rythorpe — APPROVED** [review]

**Comment by rythorpe** [comment]
Thanks @gtdang!

**Pull request merged**